Media browser: NPE trying to add a picture then cancelling

### Expected behavior
When trying to add a picture to the Media Library, it should be added, or if the action is cancelled, the user should be returned to the media browser screen.

### Actual behavior
When returning to the media browser screen without picking any picture to include in the media library, the app crashes.

```03-03 10:08:59.622 25817-25817/org.wordpress.android.beta E/AndroidRuntime: FATAL EXCEPTION: main
                                                                            Process: org.wordpress.android.beta, PID: 25817
                                                                            java.lang.RuntimeException: Failure delivering result ResultInfo{who=null, request=2000, result=0, data=null} to activity {org.wordpress.android.beta/org.wordpress.android.ui.media.MediaBrowserActivity}: java.lang.NullPointerException: Attempt to invoke virtual method 'android.net.Uri android.content.Intent.getData()' on a null object reference
                                                                                at android.app.ActivityThread.deliverResults(ActivityThread.java:4089)
                                                                                at android.app.ActivityThread.handleSendResult(ActivityThread.java:4132)
                                                                                at android.app.ActivityThread.-wrap20(ActivityThread.java)
                                                                                at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1533)
                                                                                at android.os.Handler.dispatchMessage(Handler.java:102)
                                                                                at android.os.Looper.loop(Looper.java:154)
                                                                                at android.app.ActivityThread.main(ActivityThread.java:6119)
                                                                                at java.lang.reflect.Method.invoke(Native Method)
                                                                                at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:886)
                                                                                at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:776)
                                                                             Caused by: java.lang.NullPointerException: Attempt to invoke virtual method 'android.net.Uri android.content.Intent.getData()' on a null object reference
                                                                                at org.wordpress.android.ui.media.MediaBrowserActivity.onActivityResult(MediaBrowserActivity.java:285)
                                                                                at android.app.Activity.dispatchActivityResult(Activity.java:6932)
                                                                                at android.app.ActivityThread.deliverResults(ActivityThread.java:4085)
                                                                                at android.app.ActivityThread.handleSendResult(ActivityThread.java:4132) 
                                                                                at android.app.ActivityThread.-wrap20(ActivityThread.java) 
                                                                                at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1533) 
                                                                                at android.os.Handler.dispatchMessage(Handler.java:102) 
                                                                                at android.os.Looper.loop(Looper.java:154) 
                                                                                at android.app.ActivityThread.main(ActivityThread.java:6119) 
                                                                                at java.lang.reflect.Method.invoke(Native Method) 
                                                                                at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:886) 
                                                                                at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:776) 
03-03 10:08:59.626 1035-4599/? W/ActivityManager:   Force finishing activity org.wordpress.android.beta/org.wordpress.android.ui.media.MediaBrowserActivity
```

### Steps to reproduce the behavior
1. Open the Media browser by going to the first tab -> Media
2. Hit the `+` button on the top-right corner, and choose "select from gallery"
3. the native picker is presented
4. Hit the device back button
5. Crash happens

##### Tested on [Google Pixel], Android [7.1.1]

--- a/WordPress/src/main/java/org/wordpress/android/ui/media/MediaBrowserActivity.java
+++ b/WordPress/src/main/java/org/wordpress/android/ui/media/MediaBrowserActivity.java
@@ -289,7 +289,7 @@ public class MediaBrowserActivity extends AppCompatActivity implements MediaGrid
                 }
                 break;
             case RequestCodes.TAKE_PHOTO:
-                if (resultCode == Activity.RESULT_OK && data != null) {
+                if (resultCode == Activity.RESULT_OK) {
                     Uri uri = Uri.parse(mMediaCapturePath);
                     mMediaCapturePath = null;
                     queueFileForUpload(uri, getContentResolver().getType(uri));

--- a/WordPress/src/main/java/org/wordpress/android/ui/media/MediaBrowserActivity.java
+++ b/WordPress/src/main/java/org/wordpress/android/ui/media/MediaBrowserActivity.java
@@ -282,21 +282,17 @@ public class MediaBrowserActivity extends AppCompatActivity implements MediaGrid
         switch (requestCode) {
             case RequestCodes.PICTURE_LIBRARY:
             case RequestCodes.VIDEO_LIBRARY:
-                if (resultCode == Activity.RESULT_OK) {
-                    if (data != null) {
-                        Uri imageUri = data.getData();
-                        String mimeType = getContentResolver().getType(imageUri);
-                        fetchMedia(imageUri, mimeType);
-                    }
+                if (resultCode == Activity.RESULT_OK && data != null) {
+                    Uri imageUri = data.getData();
+                    String mimeType = getContentResolver().getType(imageUri);
+                    fetchMedia(imageUri, mimeType);
                 }
                 break;
             case RequestCodes.TAKE_PHOTO:
-                if (resultCode == Activity.RESULT_OK) {
-                    if (data != null) {
-                        Uri uri = Uri.parse(mMediaCapturePath);
-                        mMediaCapturePath = null;
-                        queueFileForUpload(uri, getContentResolver().getType(uri));
-                    }
+                if (resultCode == Activity.RESULT_OK && data != null) {
+                    Uri uri = Uri.parse(mMediaCapturePath);
+                    mMediaCapturePath = null;
+                    queueFileForUpload(uri, getContentResolver().getType(uri));
                 }
                 break;
             case RequestCodes.TAKE_VIDEO:

--- a/WordPress/src/main/java/org/wordpress/android/ui/media/MediaBrowserActivity.java
+++ b/WordPress/src/main/java/org/wordpress/android/ui/media/MediaBrowserActivity.java
@@ -282,9 +282,13 @@ public class MediaBrowserActivity extends AppCompatActivity implements MediaGrid
         switch (requestCode) {
             case RequestCodes.PICTURE_LIBRARY:
             case RequestCodes.VIDEO_LIBRARY:
-                Uri imageUri = data.getData();
-                String mimeType = getContentResolver().getType(imageUri);
-                fetchMedia(imageUri, mimeType);
+                if (resultCode == Activity.RESULT_OK) {
+                    if (data != null) {
+                        Uri imageUri = data.getData();
+                        String mimeType = getContentResolver().getType(imageUri);
+                        fetchMedia(imageUri, mimeType);
+                    }
+                }
                 break;
             case RequestCodes.TAKE_PHOTO:
                 if (resultCode == Activity.RESULT_OK) {

